Lib/test/test_xml_etree.py

```diff
@@ -1786,36 +1929,38 @@ def test_cyclic_gc(self):
         class Dummy:
             pass
 
-        # Test the shortest cycle: d->element->d
-        d = Dummy()
-        d.dummyref = ET.Element('joe', attr=d)
-        wref = weakref.ref(d)
-        del d
-        gc_collect()
-        self.assertIsNone(wref())
-
-        # A longer cycle: d->e->e2->d
-        e = ET.Element('joe')
-        d = Dummy()
-        d.dummyref = e
-        wref = weakref.ref(d)
-        e2 = ET.SubElement(e, 'foo', attr=d)
-        del d, e, e2
-        gc_collect()
-        self.assertIsNone(wref())
-
-        # A cycle between Element objects as children of one another
-        # e1->e2->e3->e1
-        e1 = ET.Element('e1')
-        e2 = ET.Element('e2')
-        e3 = ET.Element('e3')
-        e1.append(e2)
-        e2.append(e2)
-        e3.append(e1)
-        wref = weakref.ref(e1)
-        del e1, e2, e3
-        gc_collect()
-        self.assertIsNone(wref())
+        with self.subTest("Test the shortest cycle: d->element->d"):
```

Comment:
Unrelated changes.

Reply:
These changes helped me to debug the changes I made to the C implementation.

Reply:
Then please remove them after finishing debugging.

Reply:
My point was that they were not wholly unrelated. What is the harm in including them here?

Reply:
I agree with @GPHemsley that it's better to have subtests than multiple tests in a single test function. Although having multiple test functions also wouldn't be all too bad. :)
Yes, they are unrelated to this specific ticket, but they don't really seem big enough to merit their own ticket and PR. They somewhat match the general refactoring of lengthy tests into subtests that this change does in other places. It's at least an edge case.